### ISSUE TITLE
Question about using model on my datasets

### ISSUE THREAD

**Issue by keruoya** [LS4GAN/uvcgan2#28]
"Hello! Thank you for your fantastic work. I'm training a model with my dataset, but the fake images from the 20th epoch are completely gray. Could this be due to an error in my operations, or because I haven't pre-trained the generator and I'm not using the models you provided?"
![屏幕截图 2024-04-14 213803](https://github.com/LS4GAN/uvcgan2/assets/85153082/26488168-cfde-4c68-9822-f5ab32b9cc48)

**Comment by keruoya** [comment]
And this is my history.csv
gen_ab,gen_ba,cycle_a,cycle_b,disc_a,disc_b,idt_a,idt_b,gp_a,gp_b,epoch,time
0.41484783540169395,0.3107682885809077,0.34295186284515594,0.3679717449678315,0.22416516832179492,0.1776486219631301,0.06967297034958998,0.16064514741301536,0.005031600462583204,0.013559102652490967,20

**Comment by usert5432** [comment]
Hi @keruoya,

The history file looks ok to me. The resulting image also contains some information (I played with color thresholds in an image editor and there is a clear hidden structure in the image).

I think, it is possible that the model was not trained long enough (with epoch = 20, this may happen if the training dataset is very small)? For reference, our generator is exponentially averaged with momentum of 0.9999. So, it takes ~10000 training iterations to just overwrite the initial (random) state of the generator.

I was wondering if it would be possible to continue the training for a little bit longer and see if it get better?

Otherwise, there are are few other things one may check to investigate this issue:
a. CycleGAN translates images in both directions. Could you also check the corresponding image in the opposite direction? Is it also gray?
b. CycleGAN outputs cyclically-translated images, which are saved under `reco_a` and `reco_b` subdirectories. Could you check if those images match `real_a` and `real_b` images? And that `real_a` and `real_b` images look as expected (if not, there may be a problem with dataloaders)?

**Comment by keruoya** [comment]
Thank you for your suggestion! I will continue with the training. I also checked the fake_b, reco_a, reco_b of the 20th epoch, they are all gray like fake_a, which may also contain hidden information? Then real_a and real_b are normal, so there shouldn't be any issue with the dataloader.And here is fake_a,fake_b,reco_a,reco_b.
![image](https://github.com/LS4GAN/uvcgan2/assets/85153082/77ec7c8e-762b-4340-af40-f388c4648745)
![image](https://github.com/LS4GAN/uvcgan2/assets/85153082/a5e7483c-efc4-4726-aa99-9c74b9aad4c3)
![image](https://github.com/LS4GAN/uvcgan2/assets/85153082/5987f7b2-b2e7-4589-a118-2ca6c443f69f)
![image](https://github.com/LS4GAN/uvcgan2/assets/85153082/3774a1f2-786e-4dea-80a1-8711b6dfe487)

**Comment by keruoya** [comment]
And the size of my datasets is 4000,is this too small?Thank you for your suggestion!

**Comment by usert5432** [comment]
Hi @keruoya,

Thank you for elaboration!

> And the size of my datasets is 4000,is this too small?

No, this should be more than enough (assuming 4000 training samples).

Looking over your images, I do see some structure in both `fake_a` and `fake_b`, but no structure in `reco_a` and `reco_b`. In my experience, when the generators fail to recover `reco_a` and `reco_b` -- it usually helps to increase `lambda_a` and `lambda_b` parameters of model.

So, I would recommend increasing these values by a factor of 10 (or 100, depending on the dataset) and trying the training again. If that still does not help, another thing that may be helpful is replacing `normal` weight initialization of the generators by the `kaiming` one.

**Comment by keruoya** [comment]
Hello, I reduced the values of lamda_a and lamda_b by a factor of 10, and it indeed worked. I generated normal images. Thank you! Although the images generated after 500 epochs visually appear better than those from CycleGAN, the PSNR and SSIM are lower compared to CycleGAN (because I have real images in domain B, so I can measure this). What should I do? Should I pre-train the generator, continue training, or adjust the values of lamda?

**Comment by Squishedmac** [comment]
hey just wanted to know if any of the above mentioned steps helped to produce a colour accurate image and not just a gray one, i have the same issue and tweaking lambda_a and lambda_b just made hte gen_ab and gen_ba loss explode up

**Comment by keruoya** [comment]
hi, @Squishedmac I just replace weight initialization of the generators by the one.normal kaiming and continue training,and here is my loss 
![image](https://github.com/LS4GAN/uvcgan2/assets/85153082/20f79ab8-2f36-4f3f-8f43-aa161e1b7a94)
